contracts/base/LendingPool.sol

```diff
@@ -423,4 +449,50 @@ contract LendingPool is Base, LendingPoolInterface {
         bool mintResult = tToken.mint(to, amount);
         require(mintResult, "TTOKEN_MINT_FAILED");
     }
+
+    /**
+        @notice It swaps the lending pool's accumualted COMP for the lendingToken
+        using Uniswap and deposits it into Compound.
+     */
+    function swapAccumulatedComp() external {
```

Comment:
Lets make this `internal`, return the value of **lendingTokens** claimed, and call it from both `_depositToCompound` and `_withdrawFromCompound`

1. When called from `_depositToCompound`, add the returned value of tokens claimed to the value of tokens that we wanted to deposit originally.
2. When called from `_withdrawFromCompound`, compare the returned valued of claimed tokens to the amount of tokens we are wanting to withdraw. 
    1. If we want to withdraw less than the amount claimed, call `_depositToCompound` for the remainder 
    2. If more than, proceed to withdraw the remaining value from compound
3. Check the balance of claimed tokens. Only swap if we actually claimed anything.
4. Can remove the `increaseSupply` call on market state since its being updated after the `_deposit` and `_withdraw` functions are called.

Reply:
You're going to have to do 1. and 2. because I don't know how the other functions work or what you want to do with this.